src/entry.rs

```diff
@@ -705,9 +719,19 @@ impl<'a> EntryFields<'a> {
             dst: &Path,
             f: Option<&mut std::fs::File>,
             mode: u32,
-            _preserve: bool,
+            preserve: bool,
         ) -> io::Result<()> {
-            Err(io::Error::new(io::ErrorKind::Other, "Not implemented"))
+            if cfg!(target_os = "wasi") {
```

Comment:
Like above I don't think that this needs specific logic for wasi/wasm, if the goal is to get a better error message then I think this can improve the error message in-line and there's no need to do something different for wasi vs other wasm targets.

Reply:
Unfortunately this is not like the other one above because of this...
https://github.com/rust-lang/rust/blob/master/library/std/src/sys/wasi/fs.rs#L538

Basically without the check with an error message the process will panic rather than return a graceful error - I think the majority of consumers would prefer an error message over a panic.

Reply:
I think this would be best handled with a new configuration option similar to preserving mtimes to preserving permissions which is on-by-default but off-by-default for wasi.

Reply:
It already has a preserve flag on line 719 which itself comes from the preserve_permissions which indeed is already following the "off-by-default for wasi" logic which I implemented in the last commit